wireapp/wire-android
# Refactor notifications to use Scala signals directly

Moves Java NotificationsController into 3, smaller scala-based ones and uses data from the `NotificationService` directly. 

Note, this pull request also refactors some of the scala classes and removes annotations processing, since we should hopefully no longer want to add any more Java controllers.

#### Test Description: 
* Ensure that all notifications come in correctly for different message types
* Ensure that saving images displays the correct notification
* Ensure that incoming and outgoing calls display the correct notifications and that the controls for them work properly

@Robin - please also check that AUDIO and VIDEO messages produce notifications now 
#### APK
[Download build #7559](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7559/artifact/build/artifact/wire-dev-PR137-7559.apk)
[Download build #7561](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7561/artifact/build/artifact/wire-dev-PR137-7561.apk)
[Download build #7563](http://192.168.10.18:8080/job/Pull%20Request%20Builder/7563/artifact/build/artifact/wire-dev-PR137-7563.apk)
